packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,355 @@
+            lfFile.encoding = args[2].encoding;
+        }else{
+            lfFile.zosResp = await Download.dataSet(...args);
+        }
+
+        if (useStash){
+            await this.destroyTempFile(path.posix.join(tmpdir(), "toDelete.txt"));
+        }
+        return lfFile;
+    }
+
+    /**
+     * Performs appropriate file comparison (either in browser or as a terminal diff) between lf-USS or lf-DS.
+     * Local file (lf) will then be opened in default editor
+     * @param {AbstractSession} session - the session object generated from the connected profile
+     * @param {IHandlerParameters} commandParameters - parameters supplied by args
+     * @returns {Promise<IZosFilesResponse>} - the response generated by {@link CompareBaseHelper.getResponse}
+     * @memberof EditUtilities
+     */
+    public static async fileComparison(session: AbstractSession, commandParameters: IHandlerParameters): Promise<IZosFilesResponse>{
+        const handlerDs = new LocalfileDatasetHandler;
+        const handlerUss = new LocalfileUssHandler;
+        const helper = new CompareBaseHelper(commandParameters);
+        const gui = ProcessUtils.isGuiAvailable();
+        const options: IDiffNameOptions = {
+            name1: "local file",
+            name2: "remote file"
+        };
+
+        helper.browserView = (gui === GuiResult.GUI_AVAILABLE);
+
+        const lf: Buffer = await handlerDs.getFile1(session, commandParameters.arguments, helper);
+        let mf: string | Buffer;
+        try{
+            if (commandParameters.positionals[2].includes('d')){
+                mf = await handlerDs.getFile2(session, commandParameters.arguments, helper);
+            }else{
+                mf = await handlerUss.getFile2(session, commandParameters.arguments, helper);
+            }
+        }catch(err){
+            throw new ImperativeError({
+                msg: TextUtils.chalk.red(`Command terminated. Issue retrieving files for comparison.`),
+                causeErrors: err
+            });
+        }
+
+        const diffResponse = await helper.getResponse(helper.prepareContent(lf), helper.prepareContent(mf), options);
+        if (!helper.browserView){
+            if (diffResponse){
+                commandParameters.response.console.log('\n'+diffResponse.commandResponse);
+            }else{
+                throw new ImperativeError({
+                    msg: TextUtils.chalk.red(`Diff was unable to be generated`)
+                });
+            }
+        }
+        return diffResponse;
+    }
+
+    /**
+     * Enable user to make their edits and wait for user input to indicate editing is complete
+     * @param {string} tempPath - parameters supplied by args
+     * @param {string} editor - optional parameter originally supplied by args
+     * @memberof EditUtilities
+     */
+    public static async makeEdits(tempPath: string, editor?: string): Promise<void>{
+        if (editor){
```

Comment:
Should this `if` block be removed? It's fine if the `editor` param passed to `ProcessUtils.openInEditor` is undefined - that just means to use the default editor.